lib/logstasher/railtie.rb

```diff
@@ -41,6 +41,20 @@ class Railtie < Rails::Railtie
     end
   end
 
+  def default_source
+    case RUBY_PLATFORM
+    when /darwin/
+      # NOTE: MacOS Sierra and later are setting `.local`
+      # hostnames that even as real hostnames without the `.local` part,
+      # are still unresolvable. One reliable way to get an IP is to
+      # get all available IP address lists and use the first one.
+      # This will always be `127.0.0.1`.
+      Socket.ip_address_list.first
+    else
+      IPSocket.getaddress(Socket.gethostname)
```

Comment:
Depending on the OS and configuration, this will return a different IP.

On my CentOS host it returns IPv6 loopback address (`::1`). If you don't have IPv6 configured, it will return `127.0.0.1`. This is not enough to identify a machine.
On my FreeBSD host it returns the external IPv6 address which is better, but still hard to read.

IMHO if you have multiple machines and you want to distinguish between logs from either, you should use the the machines hostname, or set the `source` option explicitly.